### ISSUE TITLE
avreader does not work (with sam)

### ISSUE THREAD

**Issue by cbenhagen** [tuttleofx/TuttleOFX#466]
```
OFX_PLUGIN_PATH=/usr/local/opt/tuttleofx/OFX sam do --verbose 4 tuttle.avreader oceans.mp4 // tuttle.viewer
[Process render] start           ] 0/1116 - 00:00:00
2015-08-16 11:09:54,504 - SAM ERROR - Tuttle graph computation failed.
2015-08-16 11:09:54,504 - SAM DEBUG - /tmp/tuttleofx20150814-35919-3c8vny/libraries/tuttle/src/tuttle/host/ofx/OfxhImageEffectNode.cpp(653): Throw in function virtual void tuttle::host::ofx::imageEffect::OfxhImageEffectNode::beginSequenceRenderAction(OfxTime, OfxTime, double, bool, OfxPointD)
Dynamic exception type: boost::exception_detail::clone_impl<tuttle::host::ofx::OfxhException>
[tuttle::exception::tag_backtraceMessage*] =
kOfxStatFailed
2015-08-16 11:09:54,504 - SAM INFO - Memory usage: 67KB
```

I am using version 12.1 on OSX

**Comment by cchampet** [comment]
Can you provide some information about your input file oceans.mp4?
You can give use the output of ffprobe from ffmpeg.
Or avmeta from avtranscoder (avtranscoder can be installed with homebrew):
```
brew install avtranscoder
avmeta oceans.mp4
```

**Comment by cchampet** [comment]
And one more question: did you success to read this media with a previous version of tuttle?

**Comment by cbenhagen** [comment]
http://vjs.zencdn.net/v/oceans.mp4

In fact it does not matter which file i take.. this was just an example. It
has been quite some time that i last used tuttleofx in production. But id
love to change that.

2015-08-17 10:22 GMT+02:00 Clément Champetier <notifications@github.com>:

**Comment by cchampet** [comment]
It works on my computer...
Can you check this command:
```
 sam do r oceans.mp4 v_threads=1 // viewer  --verbose trace
```

**Comment by nprabhanjan** [comment]
I tried running avreader using below sam command and it is not working.

$ sam do --verbose 4 avreader sample.mp4 v_threads=1 // avwriter sample_read.mp4 
[Process render] start           ] 0/24 - 00:00:00
        >-- /vagrant/sample.mp4  ] 1/24 - 00:00:00
        --> /vagrant/sample_read.mp4
[Process render] Stopped at frame 0.

[################################] 1/24 - 00:00:00
[Ofxh Image] The PLUGIN has not properly released the image ref count (1) of clip "TuttleAVWriter_1.Output" at time 0.
2015-08-25 16:42:11,555 - SAM ERROR - Tuttle graph computation failed.
2015-08-25 16:42:11,556 - SAM INFO - Memory usage: 113KB

**Comment by cchampet** [comment]
It seems to append with h264 video. Can you confirm @nprabhanjan ?

**Comment by nprabhanjan** [comment]
Input is mpeg2 video 422.
ffmpeg info shows below info
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'sample.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    encoder         : Lavf56.36.100
  Duration: 00:00:01.00, start: 0.000000, bitrate: 8787 kb/s
    Stream #0:0(eng): Video: mpeg2video (4:2:2) (mp4v / 0x7634706D), yuv422p(tv), 1920x1080 [SAR 1:1 DAR 16:9], 8779 kb/s, 24 fps, 24 tbr, 12288 tbn, 48 tbc (default)
    Metadata:
      handler_name    : VideoHandler
      encoder         : XDCAM HD422 1080p24


I am not specifying any output video codec but mentioning sample_read.mp4 as output file name

**Comment by cbenhagen** [comment]
@cchampet 
```
sam do r oceans.mp4 v_threads=1 // viewer  --verbose trace
Search plugins recursively in "/usr/local/opt/tuttleofx/OFX".
Found cached binary "/usr/local/opt/tuttleofx/OFX/AnisotropicDiffusion-1.1.ofx.bundle/Contents/MacOS/AnisotropicDiffusion-1.1.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/AudioVideo-4.3.ofx.bundle/Contents/MacOS/AudioVideo-4.3.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/BitDepth-1.0.ofx.bundle/Contents/MacOS/BitDepth-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Blur-1.0.ofx.bundle/Contents/MacOS/Blur-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/CheckerBoard-2.0.ofx.bundle/Contents/MacOS/CheckerBoard-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/ColorBars-2.0.ofx.bundle/Contents/MacOS/ColorBars-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/ColorCube-2.0.ofx.bundle/Contents/MacOS/ColorCube-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/ColorGradation-1.0.ofx.bundle/Contents/MacOS/ColorGradation-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/ColorSuppress-3.0.ofx.bundle/Contents/MacOS/ColorSuppress-3.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/ColorTransfer-2.0.ofx.bundle/Contents/MacOS/ColorTransfer-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/ColorWheel-2.0.ofx.bundle/Contents/MacOS/ColorWheel-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Component-1.0.ofx.bundle/Contents/MacOS/Component-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Constant-2.0.ofx.bundle/Contents/MacOS/Constant-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Crop-1.1.ofx.bundle/Contents/MacOS/Crop-1.1.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/CTL-2.0.ofx.bundle/Contents/MacOS/CTL-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Diff-1.0.ofx.bundle/Contents/MacOS/Diff-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Dpx-1.0.ofx.bundle/Contents/MacOS/Dpx-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Exr-3.0.ofx.bundle/Contents/MacOS/Exr-3.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Fade-1.0.ofx.bundle/Contents/MacOS/Fade-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Flip-1.0.ofx.bundle/Contents/MacOS/Flip-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/FloodFill-1.0.ofx.bundle/Contents/MacOS/FloodFill-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Gamma-1.0.ofx.bundle/Contents/MacOS/Gamma-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Histogram-1.0.ofx.bundle/Contents/MacOS/Histogram-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/HistogramKeyer-1.0.ofx.bundle/Contents/MacOS/HistogramKeyer-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/IdKeyer-1.0.ofx.bundle/Contents/MacOS/IdKeyer-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/ImageMagick-1.0.ofx.bundle/Contents/MacOS/ImageMagick-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/ImageStatistics-2.0.ofx.bundle/Contents/MacOS/ImageStatistics-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Invert-1.0.ofx.bundle/Contents/MacOS/Invert-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Jpeg-1.0.ofx.bundle/Contents/MacOS/Jpeg-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Jpeg2000-1.5.ofx.bundle/Contents/MacOS/Jpeg2000-1.5.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/LensDistort-2.2.ofx.bundle/Contents/MacOS/LensDistort-2.2.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/LocalMaxima-1.0.ofx.bundle/Contents/MacOS/LocalMaxima-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/MathOperator-1.0.ofx.bundle/Contents/MacOS/MathOperator-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/MemoryBuffer-1.0.ofx.bundle/Contents/MacOS/MemoryBuffer-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Merge-1.0.ofx.bundle/Contents/MacOS/Merge-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/NlmDenoiser-2.3.ofx.bundle/Contents/MacOS/NlmDenoiser-2.3.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Normalize-1.0.ofx.bundle/Contents/MacOS/Normalize-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/OCIO-1.0.ofx.bundle/Contents/MacOS/OCIO-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/OpenImageIO-1.0.ofx.bundle/Contents/MacOS/OpenImageIO-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Pinning-1.0.ofx.bundle/Contents/MacOS/Pinning-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Png-1.0.ofx.bundle/Contents/MacOS/Png-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/PushPixel-1.2.ofx.bundle/Contents/MacOS/PushPixel-1.2.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Ramp-2.0.ofx.bundle/Contents/MacOS/Ramp-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Raw-2.0.ofx.bundle/Contents/MacOS/Raw-2.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Resize-1.0.ofx.bundle/Contents/MacOS/Resize-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/SeExpr-1.0.ofx.bundle/Contents/MacOS/SeExpr-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Sobel-1.0.ofx.bundle/Contents/MacOS/Sobel-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Text-6.0.ofx.bundle/Contents/MacOS/Text-6.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Thinning-1.0.ofx.bundle/Contents/MacOS/Thinning-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/TimeShift-1.0.ofx.bundle/Contents/MacOS/TimeShift-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/TurboJpeg-1.0.ofx.bundle/Contents/MacOS/TurboJpeg-1.0.ofx"
Found cached binary "/usr/local/opt/tuttleofx/OFX/Viewer-1.0.ofx.bundle/Contents/MacOS/Viewer-1.0.ofx"
Search plugins recursively in "/Library/OFX/Plugins".
Found cached binary "/Library/OFX/Plugins/inpaint.ofx.bundle/Contents/MacOS/inpaint.ofx"
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Failed to Fetch Unknown Suite: OfxParametricParameterSuite 1.
Failed to Fetch Unknown Suite: NukeOfxCameraSuite 1.
Use "tuttle.avreader" to read "oceans.mp4".
[Process render] start           ] 0/1116 - 00:00:00
[Process render] begin timeRange: [0, 1115]
2015-08-26 19:05:41,645 - SAM ERROR - Tuttle graph computation failed.
2015-08-26 19:05:41,645 - SAM DEBUG - /tmp/tuttleofx20150814-35919-3c8vny/libraries/tuttle/src/tuttle/host/ofx/OfxhImageEffectNode.cpp(653): Throw in function virtual void tuttle::host::ofx::imageEffect::OfxhImageEffectNode::beginSequenceRenderAction(OfxTime, OfxTime, double, bool, OfxPointD)
Dynamic exception type: boost::exception_detail::clone_impl<tuttle::host::ofx::OfxhException>
[tuttle::exception::tag_backtraceMessage*] =
kOfxStatFailed
2015-08-26 19:05:41,647 - SAM INFO - Memory usage: 104KB
```

**Comment by cchampet** [comment]
Fixed with [v0.12.3](https://github.com/tuttleofx/TuttleOFX/releases/tag/v0.12.3)
@cbenhagen @nprabhanjan Can you confirm?

**Comment by cchampet** [comment]
Fell free to reopen it if it is not working with your inputs.
